### PR TITLE
Retain `WS_MAXIMZE` when unminmizing a maximized window

### PR DESCRIPTION
**Pull request by amrbashir**
- [X] Tested on all platforms changed
- [X] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [X] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [X] Created or updated an example program if it would help users understand this functionality
- [X] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

For a reproduction of the bug:
```
git clone https://github.com/amrbashir/winit --branch fix/windows/restoring-minimized-maximized
cd winit
git checkout cf92e881526767db13f3549ac091986205a213f6
cargo run --example window_debug
```
and follow the instructions displayed on the terminal

### CODE REVIEW

**Review by msiglreith — COMMENTED** [review]
I looked a bit locally into this as I would like to avoid the extra flag.
The style change in `apply_diff` seems to cause the issue with the maximized state - `MINIMIZED` and `MAXIMIZED` are handled beforehands tho using the `ShowWindow` calls. Removing the corresponding flags from the `diff` seems to work as well

**Comment by amrbashir** [comment]
> Removing the corresponding flags from the `diff` seems to work as well

Sorry, not quite following what you mean.

File: src/platform_impl/windows/window_state.rs [hunk 1]
```diff
@@ -114,6 +114,12 @@ bitflags! {
         /// Drop shadow for undecorated windows.
         const MARKER_UNDECORATED_SHADOW = 1 << 20;
 
+        /// When minmizing a maximized Windows, `WM_SIZE` which we use to update the `MAXIMIZED` bit
+        /// is fired with `wparam` set to `SIZE_MINIMIZED` and thus the `MAXIMIZED` bit will be unset
+        /// and when un-minimizing the window, the `MAXIMIZED` bit will still be unset and later on
+        /// in `apply_diff` when `new.to_window_styles()` is called, it will not add `WS_MAXIMIZE` window style.
+        const MARKER_WAS_MAXIMIZED = 1 << 18;
```
Comment (maroider) [inline src/platform_impl/windows/window_state.rs:121]: This conflicts with `IGNORE_CURSOR_EVENT`. Did you rebase without checking that the flag was still non-conflicting?

Comment (amrbashir) [reply src/platform_impl/windows/window_state.rs:121]: yeah I forgot to check. Good catch! will fix it now.

**Comment by msiglreith** [comment]
> > Removing the corresponding flags from the `diff` seems to work as well
> 
> Sorry, not quite following what you mean.

I made a commit on your reproduction commit which shows my suggestion: https://github.com/msiglreith/winit/commit/cf7f052a0872a60ac374b7f66a37de2065fb509f

**Pull request merged**